modules/apps/transfer/types/trace.go

```diff
@@ -76,15 +67,23 @@ func (dt DenomTrace) GetFullDenomPath() string {
 	return dt.GetPrefix() + dt.BaseDenom
 }
 
-// extractPathAndBaseFromFullDenom returns the trace path and the base denom from
-// the elements that constitute the complete denom.
-func extractPathAndBaseFromFullDenom(fullDenomItems []string) ([]string, string) {
+// ExtractDenomFromFullPath returns the denom from the full path.
+// Used to support v1 denoms.
+func ExtractDenomFromFullPath(fullPath string) Denom {
+	denomSplit := strings.Split(fullPath, "/")
+
+	if denomSplit[0] == fullPath {
+		return Denom{
+			Base: fullPath,
+		}
+	}
```

Comment:
https://github.com/cosmos/ibc-go/blob/main/modules/apps/transfer/types/trace_test.go#L19

Reply:
This case is caught by the `baseDenomSlice` construction https://github.com/cosmos/ibc-go/pull/6424/files#diff-ff4125d8c59b9ab8785f0ae9e950dc2a59a79428482e3f36b7b1695f57d5e39fR105

Reply:
thinking now, it's possible that the correct behaviour ends up happening later in the function here

This actually ends up being handled correctly here `baseDenomSlice = denomSplit[i:]`

I feel like we should probably still explicitly check this at the start of the fn `if denomSplit[0] == fullPath` to be more explicit. I think this edge case could slip through if the for loop code gets changed in the future. We should at least add a test like

```golang
func TestFoo(t *testing.T) {
	res := types.ExtractDenomFromFullPath("atom/")
	require.Equal(t, "atom/", res.Base)
	require.Empty(t, res.Trace)
}
```

or similar WDYT?

Reply:
When we remove `ParseDenomTrace` we can move the test cases over to directly test the extraction function (I believe it didn't have unit tests since it was a private function and written before we made use of `export_test.go`

Reply:
ah yeah I hadn't thought of additional slashes inside, I guess it's not possible to do a check at the start, maybe we can add some additional tests for this function. Happy to do this in a follow up issue. It makes me a bit uneasy to have this happening in that for loop but I think there might be no alternative :thinking: